### PR TITLE
Cnfcertsuitereport spec and status improvements

### PR DESCRIPTION
**Pull request by shirmoran**
Spec field:

- Moved "Results" fields to status field.
- Added new string fields: "certSuiteConfigRunName", "ocpVersion", "cnfCertSuiteVersion".
- Added a new object "cnf" field which contains data regarding cnf's resources.

Status field:

- Added new string fields into the TestCaseResult struct: "Reason", "Logs".
- Added a "Verdict" field.
- Added a new object "Summary" field containing total number of test cases and total number of passed\skipped\failed\errored test cases.

### CODE REVIEW

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+func NewConfig(claimContent *claim.Schema) *Config {
+	certSuiteConfigRunName := os.Getenv(podNameEnvVar)
+	reportCrName := fmt.Sprintf("%s-report", certSuiteConfigRunName)
+	cnf := newCnfSpecField(claimContent)
+
+	return &Config{
+		ReportCrName:           reportCrName,
+		Namespace:              os.Getenv(podNamespaceEnvVar),
+		CertSuiteConfigRunName: certSuiteConfigRunName,
```
Comment (greyerof) [inline cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:98]: This is wrong. We probably need another env var so the controller passes the name of the CnfCertificationSuiteRun CR that triggered the creation of this cnf certsuite pod.

Comment (shirmoran) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:98]: Though we do have an env var for it, just saved it as a variable for convenient matters (line 91) as it is used also in creating report name. Or Have I miss understood you meaning?

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	}
+}
+
+func addNodesToCnfSpecField(cnf *cnfcertificationsv1alpha1.Cnf, nodes *map[string]interface{}) {
+	for nodeName := range *nodes {
+		cnf.Nodes = append(cnf.Nodes, nodeName)
+	}
+}
+
+func addResourcesToCnfSpecField(cnf *cnfcertificationsv1alpha1.Cnf, resources *[]claim.Resource) {
```
[the remark this answers is not in the document]
Comment (shirmoran) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:67]: There is a different problem here. The hardcoded cnf.Deployment is wrong, it suppose to be a generic resource, I'll change it!

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go (hunk 1 above)
Comment (greyerof) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:98]: This field was intended to be set with the name of the CnfCertificationSuiteRun CR that triggered the process and resulted in the creation of this report CR. If you're testing with the sample in `config/samples`, it should be `cnfcertificationsuiterun-sample`:
```
$ oc get cnfcertificationsuiteruns.cnf-certifications.redhat.com -A
NAMESPACE                    NAME                              AGE   STATUS
cnf-certification-operator   cnfcertificationsuiterun-sample   16h   CertSuiteFinished
```

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+type Config struct {
+	ReportCrName           string
+	Namespace              string
+	CertSuiteConfigRunName string
+	OcpVersion             string
+	CnfCertSuiteVersion    string
+	Cnf                    cnfcertificationsv1alpha1.Cnf
+}
+
+func addNamespacesToCnfSpecField(cnf *cnfcertificationsv1alpha1.Cnf, namespaces *[]string) {
```
[the remark this answers is not in the document]
Comment (shirmoran) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:28]: Interesting! thanks

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+			Name:       pod.Name,
+			Namespace:  pod.Namespace,
+			Containers: containers,
+		})
+	}
+}
+
+func addOperatorsToCnfSpecField(cnf *cnfcertificationsv1alpha1.Cnf, csvs *[]claim.Metadata) {
+	for _, csv := range *csvs {
+		cnf.Deployments = append(cnf.Deployments, cnfcertificationsv1alpha1.CnfResource{
```
[the remark this answers is not in the document]
Comment (shirmoran) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:48]: True! haven't noticed thanks

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go (hunk 1 above)
Comment (shirmoran) [reply cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:98]: Oh right! I'll change that

File: api/v1alpha1/cnfcertificationsuitereport_types.go [hunk 5]
```diff
@@ -23,25 +23,65 @@ import (
...
+	Failed  int `json:"failed"`
+	Errored int `json:"errored"`
 }
 
 // CnfCertificationSuiteReportStatus defines the observed state of CnfCertificationSuiteReport
 type CnfCertificationSuiteReportStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// Results holds the result of each test case
```
Comment (greyerof) [inline api/v1alpha1/cnfcertificationsuitereport_types.go:81]: This text is now set as the description for the "verdict" property, which is wrong. Just remove it for now. We can create a follow-up PR to polish the type's spec so every field/property has its own description.

File: cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	cnfcertificationsv1alpha1 "github.com/greyerof/cnf-certification-operator/api/v1alpha1"
+	"github.com/greyerof/cnf-certification-operator/cnf-cert-sidecar/app/claim"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	podNameEnvVar      = "MY_POD_NAME"
+	podNamespaceEnvVar = "MY_POD_NAMESPACE"
+	cnfRunNameEnvVar   = "CNF_RUN_NAME"
```
Comment (greyerof) [inline cnf-cert-sidecar/app/cnf-cert-suite-report/cnfcertsuitereport.go:17]: I'd add the "CR" word here. Something like `RUN_CR_NAME` or `CONFIG_CR_NAME` sounds better to me. Same for the variable name: `runCrNameEnvVar` or `configCrNameEnvVar`.

**Review by greyerof — APPROVED** [review]

**Review by greyerof — APPROVED** [review]

**Pull request merged**